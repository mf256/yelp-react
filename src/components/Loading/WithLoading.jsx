/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from 'react';
import styled from 'styled-components';
import { mainColor } from '../GlobalStyles/GlobalStyles';
import Loader from './Loader';

const StyledError = styled.div`
  font-size: 1rem;
  font-weight: bold;
  margin: 4rem auto;
  text-align: center;
  border: 5px dashed ${mainColor};
  border-radius: 4px;
  padding: 0.7rem 1.7rem;
  width: 50%;
`;

function WithLoading(Component) {
  return function WihLoadingComponent({ isLoading, errorMsg, ...props }) {
    if (errorMsg) {
      return (
        <StyledError>
          {errorMsg}
        </StyledError>
      );
    }

    if (isLoading) {
      return (
        <Loader />
      );
    }

    return (<Component {...props} />);
  };
}

export default WithLoading;
