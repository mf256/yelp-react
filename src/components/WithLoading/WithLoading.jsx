/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from 'react';
import styled from 'styled-components';
import GridLoader from 'react-spinners/GridLoader';
import { mainColor } from '../GlobalStyles/GlobalStyles';

const StyledError = styled.div`
  font-size: 1rem;
  font-weight: bold;
  margin: 4rem auto;
  text-align: center;
  border: 5px dashed ${mainColor};
  border-radius: 4px;
  padding: 0.7rem 1.7rem;
  width: 50%;
`;

const StyledLoader = styled.div`
  margin: 4rem auto;
  width: 57px;
`;

function WithLoading(Component) {
  return function WihLoadingComponent({ isLoading, errorMsg, ...props }) {
    if (errorMsg) {
      return (
        <StyledError>
          {errorMsg}
        </StyledError>
      );
    }

    if (isLoading) {
      return (
        <StyledLoader>
          <GridLoader
            color={mainColor}
            loading
          />
        </StyledLoader>
      );
    }

    return (<Component {...props} />);
  };
}

export default WithLoading;
