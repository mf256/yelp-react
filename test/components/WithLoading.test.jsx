import React from 'react';
import { shallow } from 'enzyme';
import WithLoading from '../../src/components/Loading/WithLoading';

describe('WithLoading', () => {
  it('should match snapshot', () => {
    const component = shallow(<WithLoading />);

    expect(component).toMatchSnapshot();
  });

  it('should render nested component', () => {
    const testComponent = 'Test';
    const ListWithLoading = WithLoading(testComponent);
    const component = shallow(<ListWithLoading />);

    expect(component).toMatchSnapshot();
  });

  it('should render loader', () => {
    const isLoading = true;
    const testComponent = 'Test';
    const ListWithLoading = WithLoading(testComponent);
    const component = shallow(<ListWithLoading isLoading={isLoading} />);

    expect(component).toMatchSnapshot();
  });

  it('should render error message', () => {
    const isLoading = false;
    const testComponent = 'Test';
    const ListWithLoading = WithLoading(testComponent);
    const component = shallow(<ListWithLoading errorMsg="Error!" isLoading={isLoading} />);

    expect(component).toMatchSnapshot();
  });

  it('should render error message when loading', () => {
    const isLoading = true;
    const testComponent = 'Test';
    const ListWithLoading = WithLoading(testComponent);
    const component = shallow(<ListWithLoading errorMsg="Error!" isLoading={isLoading} />);

    expect(component).toMatchSnapshot();
  });
});
