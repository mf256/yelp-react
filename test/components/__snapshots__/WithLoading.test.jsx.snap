// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLoading should match snapshot 1`] = `<Component />`;

exports[`WithLoading should render error message 1`] = `
<styled.div>
  Error!
</styled.div>
`;

exports[`WithLoading should render error message when loading 1`] = `
<styled.div>
  Error!
</styled.div>
`;

exports[`WithLoading should render loader 1`] = `<Loader />`;

exports[`WithLoading should render nested component 1`] = `<Test />`;
